common/browser_context.go

```diff
@@ -413,25 +406,19 @@ func (b *BrowserContext) runWaitForEventHandler(
 		case ev := <-chEvHandler:
 			if ev.typ == EventBrowserContextClose {
 				b.logger.Debugf("BrowserContext:WaitForEvent:go():EventBrowserContextClose:return", "bctxid:%v", b.id)
-				out <- nil
-				close(out)
-
-				// We wait for one matching event only,
-				// then remove event handler by cancelling context and stopping goroutine.
-				evCancelFn()
 				return
 			}
 			if ev.typ == EventBrowserContextPage {
```

Comment:
Early return from the function here?

```go
if ev.typ != EventBrowserContextPage {
   // log and return
}

Reply:
Yep, thanks! Resolved in bbfe8df5481b1d3b0616c4699b8adb8be85bcb6e.

Reply:
Perfect, thanks!